packages/cozy-harvest-lib/src/components/OAuthService.spec.js

```diff
@@ -0,0 +1,256 @@
+  isMobileApp: jest.fn(),
+  isFlagshipApp: jest.fn()
+}))
+
+describe('OAuthService', () => {
+  describe('openOAuthWindow', () => {
+    beforeEach(() => {
+      mockPopup.clear()
+      mockCozyRealtime.clear()
+    })
+
+    afterEach(() => {
+      jest.clearAllMocks()
+    })
+
+    describe('on Flagship app', () => {
+      beforeEach(() => {
+        isFlagshipApp.mockReturnValue(true)
+      })
+
+      it('should render', async () => {
+        const webviewIntent = {
+          call: jest.fn().mockImplementation(fnName => {
+            if (fnName === 'fetchSessionCode') {
+              return Promise.resolve('somesessioncode')
+            } else if (fnName === 'showInAppBrowser') {
+              return Promise.resolve({ result: 'dismiss' })
+            } else if (fnName === 'closeInAppBrowser') {
+              return Promise.resolve()
+            }
+          })
+        }
+        const promise = openOAuthWindow({
+          client,
+          konnector: { slug: 'ameli' },
+          redirectSlug: 'home',
+          extraParams: { extraParam1: 'extraParamValue' },
+          title: 'popup title',
+          account: { _id: 'someaccountid' },
+          webviewIntent
+        })
+
+        expect(isPending(promise)).toBeTruthy()
+        expect(terminateOAuth).not.toHaveBeenCalled()
+
+        mockCozyRealtime.emitRealtimeEvent(
+          'notified',
+          'io.cozy.accounts',
+          OAUTH_REALTIME_CHANNEL,
+          {
+            _id: 'oauth-popup',
+            data: {
+              error: null,
+              finalLocation: 'connection_id=194&state=someuuid',
+              key: null,
+              oAuthStateKey: 'someuuid'
+            }
+          }
+        )
+        const result = await promise
+        expect(result).toStrictEqual({
+          result: 'OK',
+          key: null,
+          data: {
+            error: null,
+            finalLocation: 'connection_id=194&state=someuuid',
+            key: null,
+            oAuthStateKey: 'someuuid'
+          }
+        })
+
+        expect(terminateOAuth).toHaveBeenCalled()
+      })
+    })
+
+    describe('with intentsApi', () => {
+      beforeEach(() => {
+        isMobileApp.mockReturnValue(true)
+      })
+
+      it('should render', async () => {
+        let closePromise = new Promise(() => {})
+        const intentsApi = {
+          fetchSessionCode: jest.fn().mockResolvedValue('somesessioncode'),
+          showInAppBrowser: jest.fn().mockImplementation(() => {
+            return closePromise
+          }),
+          closeInAppBrowser: jest.fn().mockResolvedValue()
+        }
+        const promise = openOAuthWindow({
+          client,
+          konnector: { slug: 'ameli' },
+          redirectSlug: 'home',
+          extraParams: { extraParam1: 'extraParamValue' },
+          title: 'popup title',
+          account: { _id: 'someaccountid' },
+          intentsApi
+        })
+
+        expect(isPending(promise)).toBeTruthy()
```

Comment:
Il me semble qu’on évite les truthy et qu’on privilégie les to.be.true @JF-Cozy right?

Reply:
tutafé